java/client/src/main/java/glide/api/commands/StringBaseCommands.java

```diff
@@ -122,6 +122,27 @@ public interface StringBaseCommands {
      */
     CompletableFuture<String> set(String key, String value, SetOptions options);
 
+    /**
+     * Sets the given key with the given value. Return value is dependent on the passed options.
+     *
+     * @see <a href="https://redis.io/commands/set/">redis.io</a> for details.
+     * @param key The key to store.
+     * @param value The value to store with the given key.
+     * @param options The Set options.
+     * @return Response from Redis containing a <code>String</code> or <code>null</code> response. If
+     *     the value is successfully set, return <code>"OK"</code>. If value isn't set because of
+     *     {@link ConditionalSet#ONLY_IF_EXISTS} or {@link ConditionalSet#ONLY_IF_DOES_NOT_EXIST}
+     *     conditions, return <code>null</code>. If {@link SetOptionsBuilder#returnOldValue(boolean)}
+     *     is set, return the old value as a <code>String</code>.
```

Comment:
That's a possible issue- if the previous value also stored as bytes

Reply:
This is more than a "comment" mistake. Because this function might return the old value, then we should change the function the return `byte[]` as well:

```java
CompletableFuture<byte[]> set(byte[] key, byte[] value, SetOptions options);
```

Reply:
Yep, this is what I wrote....